src/state/store.ts

```diff
@@ -0,0 +1,28 @@
+import { Action, combineReducers, configureStore, Dispatch } from '@reduxjs/toolkit';
+import { legendReducer } from './legendSlice';
+
+const rootReducer = combineReducers({
+  // tooltip: tooltipReducer,
+  legend: legendReducer,
+});
+
+export const createStore = (chartName: string = 'Chart') => {
+  console.log('createStore', chartName);
+  return configureStore({
+    reducer: rootReducer,
+    middleware: getDefaultMiddleware =>
+      getDefaultMiddleware({
+        serializableCheck: {
+          ignoredActions: ['legend/addSupplier', 'legend/removeSupplier'],
```

Comment:
I'm considering turning this off completely - Recharts has a habit of passing everything everywhere so we might as well allow non-serializables in the state straight away.